WooCommerce/Classes/POS/Presentation/Card Present Payments/Payment Alerts/CardPresentPaymentAlertViewModel.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+
+enum CardPresentPaymentAlertViewModel {
+    case scanningForReaders(CardPresentPaymentScanningForReadersAlertViewModel)
+    case scanningFailed(CardPresentPaymentScanningFailedAlertViewModel)
+
+    case foundReader(CardPresentPaymentFoundReaderAlertViewModel)
+
+    case updatingReader(CardPresentPaymentUpdatingReaderAlertViewModel)
+    case updateFailed(CardPresentPaymentReaderUpdateFailedAlertViewModel)
+
+    case connectingToReader(CardPresentPaymentConnectingToReaderAlertViewModel)
+    case connectingFailed(CardPresentPaymentConnectingFailedAlertViewModel)
+
+    case preparingForPayment(CardPresentPaymentPreparingForPaymentAlertViewModel)
+
+    case tapSwipeOrInsertCard(CardPresentPaymentTapSwipeInsertCardAlertViewModel)
+
+    case processing(CardPresentPaymentProcessingPaymentAlertViewModel)
+    case displayReaderMessage(CardPresentPaymentDisplayReaderMessageAlertViewModel)
+
+    case success(CardPresentPaymentSuccessAlertViewModel)
+
+    case error(CardPresentPaymentErrorAlertViewModel)
+}
+
+extension CardPresentPaymentAlertDetails {
+    func toAlertViewModel() -> CardPresentPaymentAlertViewModel {
+        switch self {
+        case .scanningForReaders(let endSearch):
+                .scanningForReaders(CardPresentPaymentScanningForReadersAlertViewModel(endSearchAction: endSearch))
+
+        case .scanningFailed(let error, let endSearch):
+                .scanningFailed(CardPresentPaymentScanningFailedAlertViewModel())
+
+        case .bluetoothRequired:
+                .scanningFailed(CardPresentPaymentScanningFailedAlertViewModel())
+
+        case .connectingToReader:
+                .connectingToReader(CardPresentPaymentConnectingToReaderAlertViewModel())
+
+        case .connectingFailed(let error, let retrySearch, let endSearch):
+            .connectingFailed(CardPresentPaymentConnectingFailedAlertViewModel())
+
+        case .connectingFailedNonRetryable(let error, let endSearch):
+            .connectingFailed(CardPresentPaymentConnectingFailedAlertViewModel())
+
+        case .connectingFailedUpdatePostalCode(let retrySearch, let endSearch),
+                .connectingFailedChargeReader(let retrySearch, let endSearch):
+            .connectingFailed(CardPresentPaymentConnectingFailedAlertViewModel())
+
+        case .connectingFailedUpdateAddress(let wcSettingsAdminURL, let retrySearch, let endSearch):
+            .connectingFailed(CardPresentPaymentConnectingFailedAlertViewModel())
```

Comment:
It's nice that we have the option to group different failure cases to show the same UI 👍

Reply:
Yeah. At the moment this is just for laziness/stopping the PR getting too big though!

I'll replace all these with individual views/view models when I actually implement them properly, I think.